java/operator/src/main/java/com/instaclustr/cassandra/operator/controller/DataCenterReconciliationController.java

```diff
@@ -116,6 +116,17 @@ private ConfigMapVolumeMount(final String name, final String mountPath, final V1
         }
     }
 
+    private V1PodSpec createPodSpec() {
```

Comment:
This looks like a partial refactor. This PR puts creating the `V1PodSpec` instance and assigning optional secrets, but leaves the rest of the pod spec creation code in `createOrReplaceStateNodesStatefulSet`.

It should be all in the same spot. Either we pull out the entire pod spec creation code into this function, or we remove this function and keep the code in `createOrReplaceStateNodesStatefulSet`.

Reply:
Note that we're already doing optional changes to the pod spec in `createOrReplaceStateNodesStatefulSet`.